ibc_tests/transfer_test.go

```diff
@@ -0,0 +1,170 @@
+
+	_, err = suite.chainB.SendMsgs(msg)
+	suite.Require().NoError(err) // message committed
+
+	// relay send
+	// NOTE: fungible token is prefixed with the full trace in order to verify the packet commitment
+	fullDenomPath := types.GetPrefixedDenom(pathBtoC.EndpointB.ChannelConfig.PortID, pathBtoC.EndpointB.ChannelID, voucherDenomTrace.GetFullDenomPath())
+	fungibleTokenPacket = types.NewFungibleTokenPacketData(voucherDenomTrace.GetFullDenomPath(), coinSentFromAToB.Amount.String(), suite.chainB.SenderAccount.GetAddress().String(), suite.chainC.SenderAccount.GetAddress().String())
+	packet = channeltypes.NewPacket(fungibleTokenPacket.GetBytes(), 1, pathBtoC.EndpointA.ChannelConfig.PortID, pathBtoC.EndpointA.ChannelID, pathBtoC.EndpointB.ChannelConfig.PortID, pathBtoC.EndpointB.ChannelID, timeoutHeight, 0)
+	err = pathBtoC.RelayPacket(packet, ack.Acknowledgement())
+	suite.Require().NoError(err) // relay committed
+
+	coinSentFromBToC := sdk.NewCoin(types.ParseDenomTrace(fullDenomPath).IBCDenom(), amount)
+	balance = GetLikeApp(suite.T(), suite.chainC).BankKeeper.GetBalance(suite.chainC.GetContext(), suite.chainC.SenderAccount.GetAddress(), coinSentFromBToC.Denom)
+
+	// check that the balance is updated on chainC
+	suite.Require().Equal(coinSentFromBToC, balance)
+
+	// check that balance on chain B is empty
+	balance = GetLikeApp(suite.T(), suite.chainB).BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), coinSentFromBToC.Denom)
+	suite.Require().Zero(balance.Amount.Int64())
+
+	// send from chainC back to chainB
+	msg = types.NewMsgTransfer(pathBtoC.EndpointB.ChannelConfig.PortID, pathBtoC.EndpointB.ChannelID, coinSentFromBToC, suite.chainC.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String(), timeoutHeight, 0)
+
+	_, err = suite.chainC.SendMsgs(msg)
+	suite.Require().NoError(err) // message committed
+
+	// relay send
+	// NOTE: fungible token is prefixed with the full trace in order to verify the packet commitment
+	fungibleTokenPacket = types.NewFungibleTokenPacketData(fullDenomPath, coinSentFromBToC.Amount.String(), suite.chainC.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String())
+	packet = channeltypes.NewPacket(fungibleTokenPacket.GetBytes(), 1, pathBtoC.EndpointB.ChannelConfig.PortID, pathBtoC.EndpointB.ChannelID, pathBtoC.EndpointA.ChannelConfig.PortID, pathBtoC.EndpointA.ChannelID, timeoutHeight, 0)
+	err = pathBtoC.RelayPacket(packet, ack.Acknowledgement())
+	suite.Require().NoError(err) // relay committed
+
+	balance = GetLikeApp(suite.T(), suite.chainB).BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), coinSentFromAToB.Denom)
+
+	// check that the balance on chainA returned back to the original state
+	suite.Require().Equal(coinSentFromAToB, balance)
```

Comment:
balance was got from chainB in line 153, instead of chainA stated in comment?
Also checking original, we should check `coinToSendToB` instead of `coinSentFromAToB` ?

Reply:
Same as below, probably meant endpointA (ChainB), let's update the comment

checking `coinSentFromAToB` is expected for chainB pov (it has ibc port and channel id prefix in the denom); `coinToSendToB` has the base denom & should be used for chainA pov only?

Reply:
for more info, the test case's flow is A's token to B, then further forward to C, and finally transfer back to B

Reply:
yeah so just the comment confuses me